#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.12211603774783164 0.13119482611140781 -0.9838066837170643 1.615588382820396
  position 0.07910079536859621 1.4725216504657357 0.44983809488263193
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0 0.05 0
}
DEF SCARA Robot {
  translation -1.3021682010346342e-08 0.0003306596451992606 0.12668906692837625
  rotation -0.9999999981152357 -5.755100370244266e-05 -2.1387168654933285e-05 0.0025739905513188586
  scale 1.28457 1.28457 1.28457
  children [
    Solid {
      translation 0 0 -0.03
      children [
        DEF ARM HingeJoint {
          jointParameters HingeJointParameters {
            position -9.823551264182659e-07
            axis 0 0 1
            anchor 0 0.025 0.06
          }
          device [
            RotationalMotor {
              name "armMotor"
              acceleration 3
              maxVelocity 0.5
            }
            PositionSensor {
              name "armPositionSensor"
            }
          ]
          endPoint Solid {
            translation -4.9117799999999675e-09 0.09999999999999953 0.075
            rotation -0.9830652627004953 -0.00180262779822917 -0.18324693668552147 5.362016749706212e-06
            children [
              DEF ENDEFFECTOR HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 0 1
                  anchor 0 0.08 0.04
                }
                device [
                  RotationalMotor {
                    name "headMotor"
                    acceleration 3
                    maxVelocity 0.5
                    minPosition -2.5
                    maxPosition 2.5
                  }
                ]
                endPoint Solid {
                  translation -2.0816681711721685e-17 0.15999999999999998 0.04
                  children [
                    DEF head Solid {
                      children [
                        DEF operator Solid {
                          translation 0 0.05 -0.04
                          children [
                            GPS {
                            }
                            DEF operatorSensor1 DistanceSensor {
                              translation 0.02 0 0
                              children [
                                DEF opSensor1 Shape {
                                  appearance PBRAppearance {
                                    roughness 1
                                    metalness 0
                                  }
                                  geometry Box {
                                    size 0.01 0.01 0.01
                                  }
                                  castShadows FALSE
                                }
                              ]
                              name "operatorSensor1"
                            }
                            DEF operatorSensor2 DistanceSensor {
                              translation -0.02 0 0
                              rotation 0 0 1 3.141592653589793
                              children [
                                DEF opSensor2 Shape {
                                  appearance PBRAppearance {
                                    roughness 1
                                    metalness 0
                                  }
                                  geometry Box {
                                    size 0.01 0.01 0.01
                                  }
                                  castShadows FALSE
                                }
                              ]
                              name "operatorSensor2"
                            }
                            DEF operator Shape {
                              appearance PBRAppearance {
                                baseColor 1 0 0
                                roughness 1
                                metalness 0
                              }
                              geometry Box {
                                size 0.03 0.03 0.03
                              }
                              castShadows FALSE
                            }
                          ]
                          name "operator"
                        }
                        DEF head Shape {
                          appearance PBRAppearance {
                            roughness 1
                            metalness 0
                          }
                          geometry Box {
                            size 0.1 0.2 0.05
                          }
                          castShadows FALSE
                        }
                      ]
                      name "head"
                    }
                  ]
                  name "solid(1)"
                  boundingObject USE head
                }
              }
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 1
                  roughness 1
                }
                geometry Box {
                  size 0.08 0.2 0.03
                }
              }
            ]
            boundingObject Transform {
              children [
                Box {
                  size 0.08 0.2 0.03
                }
              ]
            }
          }
        }
        DEF body Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.12
            radius 0.05
          }
          castShadows FALSE
        }
      ]
      name "Body"
    }
  ]
  name "SCARA"
  locked TRUE
  controller "arm"
  supervisor TRUE
}
